ykrt/pass_finder/try_passes.py

```diff
@@ -185,12 +228,111 @@ def binary_split(logf, passes, is_prelink):
         print(72 * "=")
         print(list_of_passes_to_str(ok_passes))
 
+def evaluate_fitness(logf, is_prelink, entity, passes):
+    try_passes = []
+    for i, bit in enumerate(entity):
+        if bit == 1:
+            try_passes.append(passes[i])
+
+    config = get_pipeline_config(is_prelink, try_passes)
+    ret, exec_time = test_pipeline(logf, config)
+    nFlags = len(try_passes) 
+    if ret:
+        try:
+            exec_time = float(exec_time)  # Convert exec_time to a float
+            # Return True for successful programs and execution time as a negative value
+            print(f"\033[92m returned {ret} and execution time {exec_time}\033[0m")
+            return  [nFlags, exec_time]
+        except ValueError:
+            print(f"Error converting exec_time to float: {exec_time}")
```

Comment:
Is there a reason for catching this error? Why wouldn't we be able to convert the time to a float?

Reply:
I am converting the exec_time from `str` to `float` so this check makes sure if the converstion is wrong we get a message.

Reply:
Have you ever seen it go wrong? If so, that probably means we're in deep trouble! Basically I'm not a fan of defensive-because-we-don't-know programming. So I would prefer to just say `float(x)` and if it raises an exception, then we know we have a bug to fix, whereas the `try/except` is likely to hide programming mistakes.

Reply:
ah, I see. Then it makes sense to remove `try/except` :)